packages/graphiql/src/main.tsx

```diff
@@ -26,24 +26,28 @@ import './css/doc-explorer.css';
 import './css/history.css';
```

Comment:
I'm not sure how this file ended up in `src` (maybe I moved it?) but it should not be part of the resulting build. It's meant to just be an example implementation. It should be `example.js`, and not be a typescript file. The bundler should not pick up this file

Reply:
Yes, it seems that I didn't notice that tsc will compile all the files in the directory, and src/main.tsx is only used as the entry script of the web application (the official practice of the vite project). Probably should be excluded from tsc or moved to resources/ directory

Reply:
ah, yes `src/main.tsx` should be `src/example.js` and should not be used as the bundling entry point, that should be `src/cdn.js` if you see in the webpack config. I originally had this file in the `resources/` directory, it must have been moved during the tabs effort.  `src/index.js` is the ESM main, which is different, and not what you want to treat as the entrypoint with vite.

Reply:
Fixed, I excluded `src/main.tsx` from `tsconfig.json/tsconfig.esm.json`

Reply:
this file should not be importing individual css files either, the idea is that it's supposed to show a demo using the CDN bundle and thus the css bundle... I need to clean this file up in another PR!

Reply:
@rxliuli is it using `src/cdn.tsx` as an entry point now? I don't see the config for it

Reply:
`build: { lib: { entry: 'src/cdn.js', name: 'GraphiQL', formats: ['es', 'umd'] } }`

Reply:
I am not using `src/cdn.tsx` as entry file

Reply:
Understood, I also need to bundle the content of esm/umd, I will add some configuration

Reply:
An error occurred, code splitting is not supported in umd format

```js
Invalid value "umd" for option "output.format" - UMD and IIFE output formats are not supported for code-splitting builds.
error during build:
Error: Invalid value "umd" for option "output.format" - UMD and IIFE output formats are not supported for code-splitting builds.
    at error (C:\Users\rxliuli\Code\Pkg\graphiql\node_modules\rollup\dist\shared\rollup.js:158:30)
    at validateOptionsForMultiChunkOutput (C:\Users\rxliuli\Code\Pkg\graphiql\node_modules\rollup\dist\shared\rollup.js:16118:16)
    at Bundle.generate (C:\Users\rxliuli\Code\Pkg\graphiql\node_modules\rollup\dist\shared\rollup.js:15957:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async handleGenerateWrite (C:\Users\rxliuli\Code\Pkg\graphiql\node_modules\rollup\dist\shared\rollup.js:23489:23)
    at async doBuild (C:\Users\rxliuli\Code\Pkg\graphiql\packages\graphiql\node_modules\vite\dist\node\chunks\dep-27bc1ab8.js:39180:26)
    at async build (C:\Users\rxliuli\Code\Pkg\graphiql\packages\graphiql\node_modules\vite\dist\node\chunks\dep-27bc1ab8.js:39011:16)
    at async CAC.<anonymous> (C:\Users\rxliuli\Code\Pkg\graphiql\packages\graphiql\node_modules\vite\dist\node\cli.js:738:9)
```

Reply:
huh weird, webpack supports it for `umd` by just un-splitting the code and inlining the dynamic imports. this is going to be a big problem then.

Reply:
the esm bundle users want dynamic imports for SSR, and the cdn users just want a single file and don't really need dynamic imports. there has to be a way?

Reply:
@rxliuli it seems some of these solutions might work:

https://github.com/vitejs/vite/issues/2982

Reply:
I played around with this too recently. There's a option that inlines all dynamic imports:
```ts
export default defineConfig({
  // ...
  build: {
    lib: { /* libarary mode */ },
    rollupOptions: {
      external: ['react', 'react-dom'],
      output: { inlineDynamicImports: true },
    },
  },
});
```

Reply:
brilliant!